test/e2e/kubernetes/kubernetes_test.go

```diff
@@ -545,6 +545,31 @@ var _ = Describe("Azure Container Cluster using the Kubernetes Orchestrator", fu
 			Expect(len(nodeList)).To(Equal(nodes))
 		})
 
+		It("should have node labels specific to masters or agents", func() {
+			nodeList, err := node.Get()
+			Expect(err).NotTo(HaveOccurred())
+			Expect(len(nodeList.Nodes)).To(Equal(eng.NodeCount()))
+			for _, node := range nodeList.Nodes {
+				labels := node.Metadata.Labels
+				Expect(labels).To(SatisfyAny(
```

Comment:
So the idea here is that *all* nodes in *all* versions on *all* OS flavors will have either `kubernetes.azure.com/role=master` or `kubernetes.azure.com/role=agent`. And then Linux nodes which are < 1.16 will additionally have either `kubernetes.io/role=master` or `kubernetes.io/role=agent` + either `node-role.kubernetes.io=master` or `node-role.kubernetes.io=agent` ?

Reply:
A nit would be: based on the first, universal test, do the follow up (if linux + < 1.16) tests in a way that ensures nodes don't have master + agent labels. So those follow-up tests would get "if master ensure add'l labels are master, if agent ensure add'l labels are agent" type thing.

Reply:
Just to clarify the state of the world in this PR:
- All nodes get `kubernetes.azure.com/role` (directly via `kubelet --node-labels`)
- All 1.15 nodes get `kubernetes.io/role` and `node-role.kubernetes.io` (directly via `kubelet --node-labels`)
- Linux 1.16 nodes get `kubernetes.io/role` and `node-role.kubernetes.io` (indirectly via the systemd service)
- Windows 1.16 nodes don't get `kubernetes.io/role` and `node-role.kubernetes.io` (need to fix this, see #1660)